### PR TITLE
call compile when generating docs (#932)

### PR DESCRIPTION
**Pull request by beckjake**
When we're doing `dbt docs generate`, implicitly call `dbt compile` unless the user provides the `--no-compile` flag.

### CODE REVIEW

File: dbt/main.py [hunk 1]
```diff
@@ -355,7 +355,21 @@ def parse_args(args):
...
+    generate_sub.set_defaults(cls=generate_task.GenerateTask,
+                              which='generate')
+    generate_sub.add_argument(
+        '--no-compile',
+        action='store_false',
+        dest='compile',
+        help='Do not run "dbt compile" as part of docs generation'
+    )
+
+    for sub in [run_sub, compile_sub, generate_sub]:
```
Comment (cmcarthur) [inline dbt/main.py]: ahhh this is a good idea! this means you can specify models to compile, or exclude models in your dbt docs generate command, right?

File: dbt/task/generate.py [hunk 2]
```diff
@@ -127,6 +128,15 @@ def _get_manifest(self):
...
 
     def run(self):
+        compile_results = None
+        if self.args.compile:
+            compile_results = super(GenerateTask, self).run()
+            if any(r.errored for r in compile_results):
+                dbt.ui.printer.print_timestamped_line(
+                    'compile failed, cannot generate docs'
+                )
+                return
```
Comment (cmcarthur) [inline dbt/task/generate.py]: did you try just returning the compile_results here? I imagine that `handle_and_check` would print out the compilation errors

Comment (beckjake) [reply dbt/task/generate.py]: I think the way I overrode it I'll have to return `{'compile_results': compile_results}`, but that's a good idea.

File: dbt/main.py (hunk 1 above)
Comment (beckjake) [reply dbt/main.py]: Yeah. You actually can't even do the `super(...).run()` without this (or adding a bunch of stuff to `self.args` manually, I suppose). But it has some nice consequences like that!

**Comment by drewbanin** [comment]
This all looks sane to me. I was hoping there was an opportunity to just do the parsing step once here, but it looks like we'll need to make some meaningful changes to parsing/compilation to make that happy.

I think it's a good thing we're using this interface (eg. `dbt docs generate [--no-compile]`). We can always optimize the underlying implementation in the future. @cmcarthur is this all good on your end?

**Pull request merged**